webapp/tests/karma/ts/modules/tasks/tasks.component.spec.ts

```diff
@@ -0,0 +1,227 @@
+    RulesEngine = {
+      isEnabled: sinon.stub().resolves(true),
+      fetchTaskDocsForAllContacts: sinon.stub().resolves([]),
+    };
+
+    Tour = sinon.stub();
+    Telemetry = { record: sinon.stub() };
+    contactTypesService = {
+      includes: sinon.stub(),
+    };
+
+    TestBed.configureTestingModule({
+      imports: [
+        TranslateModule.forRoot({ loader: { provide: TranslateLoader, useClass: TranslateFakeLoader } }),
+        RouterTestingModule,
+      ],
+      providers: [
+        provideMockStore(),
+        { provide: ChangesService, useValue: { subscribe: Changes }},
+        { provide: RulesEngineService, useValue: RulesEngine },
+        { provide: TelemetryService, useValue: Telemetry },
+        { provide: TourService, useValue: { startIfNeeded: TourService } },
+        { provide: ContactTypesService, useValue: contactTypesService }
+      ],
+      declarations: [
+        TasksComponent,
+        NavigationComponent,
+      ],
+    });
+
+    getComponent = () => {
+      return TestBed.compileComponents().then(() => {
+        fixture = TestBed.createComponent(TasksComponent);
+        component = fixture.componentInstance;
+        store = TestBed.inject(MockStore);
+        fixture.detectChanges();
+      });
+    };
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    clock?.restore();
+  });
+
+  it('should ngOnDestroy should unsubscribe and clear state', async () => {
+    await getComponent();
+
+    const setTasksList = sinon.stub(TasksActions.prototype, 'setTasksList');
+    const setTasksLoaded = sinon.stub(TasksActions.prototype, 'setTasksLoaded');
+    const spySubscriptionsUnsubscribe = sinon.spy(component.subscription, 'unsubscribe');
+    component.ngOnDestroy();
+    expect(spySubscriptionsUnsubscribe.callCount).to.equal(1);
+    expect(setTasksList.callCount).to.equal(1);
+    expect(setTasksList.args[0]).to.deep.equal([[]]);
+    expect(setTasksLoaded.callCount).to.equal(1);
+    expect(setTasksLoaded.args[0]).to.deep.equal([false]);
+  });
+
+  it('initial state before resolving tasks', async () => {
+    RulesEngine.isEnabled.callsFake(() => new Promise(() => {}));
+    await getComponent();
+    expect(component.loading).to.be.true;
+    expect(!!component.hasTasks).to.be.false;
+    expect(!!component.error).to.be.false;
+    expect(!!component.tasksDisabled).to.be.false;
+
+    expect(Tour.callCount).to.eq(0);
+  });
+
+  it('rules engine is disabled', async () => {
+    RulesEngine.isEnabled.resolves(false);
+
+    await new Promise(resolve => {
+      sinon.stub(TasksActions.prototype, 'setTasksList').callsFake(resolve);
+      getComponent();
+    });
+    expect(component.loading).to.be.false;
+    expect(!!component.hasTasks).to.be.false;
+    expect(!!component.error).to.be.false;
+    expect(component.tasksDisabled).to.be.true;
+  });
+
+  it('rules engine throws in initialization', async () => {
+    RulesEngine.isEnabled.rejects('error');
+    await new Promise(resolve => {
```

Comment:
Just curious, can we use the 'fakeasync' and 'tick' here?
Ah this is to test the effect that Rules Engine has when the component is created (?)

Reply:
Yea, the way that the old tests were built was that we would consider the "setup" done when the `setTaskList` was called. 

I found the way that this was achieved in these tests is kinda interesting, so I wanted to preserve the author's authenticity. 
I found that fakeasync doesn't work with fake timers though :(